fuzz/fuzz_targets/decode_image.rs

```diff
@@ -0,0 +1,19 @@
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+
+fuzz_target!(|data: &[u8]| {
+    let mut decoder = if let Ok(d) = tiff::decoder::Decoder::new(std::io::Cursor::new(data)) {
+        d
+    } else {
+        return;
+    };
+
+    let mut limits = tiff::decoder::Limits::default();
+    limits.decoding_buffer_size = 1_000_000;
+    limits.ifd_value_size = 1_000_000;
+    limits.intermediate_buffer_size = 1_000_000;
+
+    decoder = decoder.with_limits(limits);
+
+    let _ = decoder.read_image();
```

Comment:
This decodes exactly one image which might miss issues that appear when multiple images appear consecutively within the same file, or non-termination due to missing the end of file etc.. It would be proper to wrap this in a while-loop checking `reader.more_images()` and calling `reader.next_image` or simply iterating like so

> `while let Ok(_) = reader.next_image()`

Reply:
Pushed what I *think* is a correct implementation.

And yeah, you're right, I can find timeouts and overflows with that change. So looks like I have more fixes to do!